internal/cmd/io_binary.go

```diff
@@ -0,0 +1,276 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"github.com/Microsoft/go-winio"
+	"github.com/Microsoft/hcsshim/internal/log"
+	"github.com/containerd/containerd/namespaces"
+	"github.com/pkg/errors"
+	"io"
+	"net"
+	"net/url"
+	"os/exec"
+	"sync"
+	"time"
+)
+
+const pipeRootBinary = `\\.\pipe`
+
+func NewBinaryCmd(ctx context.Context, uri *url.URL, id string, ns string) *exec.Cmd {
+	var args[] string
+	for k, vs := range uri.Query() {
+		args = append(args, k)
+		if len(vs) > 0 {
+			args = append(args, vs[0])
+		}
+	}
+
+	execPath := "C:\\Users\\Administrator\\LCOW\\" + uri.Host
+
+	cmd := exec.CommandContext(ctx, execPath, args...)
+	cmd.Env = append(cmd.Env,
+		"CONTAINER_ID="+id,
+		"CONTAINER_NAMESPACE="+ns,
+	)
+
+	return cmd
+}
+
+type binaryIO struct {
```

Comment:
small note: this struct doesn't look like it was formatted. If you're using vs code, the go extensions will normally do this for you. Otherwise, you'll need to run `go fmt` on each file.